windshaft/driver.js

```diff
@@ -86,18 +86,20 @@ function getRequestParameters(request) {
         params.interactivity = 'uuid,occurred_from,data';
         params.style = eventsStyle;
 
-        // build query for record points
-        params.sql = baseRecordQuery;
-        if (params.id !== 'ALL') {
-            params.sql += filterRecordQuery + params.id + "'";
+        // build query for record points if do not already have a query
+        if (request.query.sql) {
+            params.sql = request.query.sql;
```

Comment:
Maybe sign the SQL with a combination of [itsdangerous](http://pythonhosted.org/itsdangerous/) and [node-jsjws](https://github.com/davedoesdev/node-jsjws)?